File: easy/find_max_consecutive_ones_test.go
package easy

import (
	"testing"
)

func TestFindMaxConsecutiveOnes(t *testing.T) {
	cases := []struct {
		num      []int
		expected int
	}{
		{[]int{1, 1, 0, 1, 1, 1}, 3},
		{[]int{1, 0, 1, 1, 0, 1}, 2},
		{[]int{0}, 0},
		{[]int{1}, 1},
		{[]int{1, 1, 1, 1}, 4},
	}
	for _, c := range cases {
		t.Run("", func(t *testing.T) {
			result := FindMaxConsecutiveOnes(c.num)
			if result != c.expected {
				t.Errorf("FindMaxConsecutiveOnes(%v) == %d, expected %v", c.num, result, c.expected)
			}

		})

	}
}

File: src/test/java/easy/FindMaxConsecutiveOnesTest.java
package easy;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.stream.Stream;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

public class FindMaxConsecutiveOnesTest {
  FindMaxConsecutiveOnes instance = new FindMaxConsecutiveOnes();

  @ParameterizedTest
  @MethodSource("dataFindMaxConsecutiveOnes")
  public void testFindMaxConsecutiveOnes(int[] nums, int expected) {

    int result = instance.solve(nums);

    assertEquals(expected, result);
  }

  public static Stream<Arguments> dataFindMaxConsecutiveOnes() {
    return Stream.of(
        Arguments.of(new int[] {1, 1, 0, 1, 1, 1}, 3),
        Arguments.of(new int[] {1, 0, 1, 1, 0, 1}, 2),
        Arguments.of(new int[] {0}, 0),
        Arguments.of(new int[] {1}, 1),
        Arguments.of(new int[] {1, 1, 1, 1}, 4));
  }
}
